packages/grafana-ui/src/components/Table/FilterPopup.tsx

```diff
@@ -0,0 +1,165 @@
+  const filteredOptions = useMemo(() => getFilteredOptions(options, column.filterValue), [options, column.filterValue]);
+  const [values, setValues] = useState<SelectableValue[]>(filteredOptions ?? []);
+
+  const onCancel = useCallback((event?: React.MouseEvent) => onClose(), [onClose]);
+
+  const onFilter = useCallback(
+    (event: React.MouseEvent) => {
+      const filtered = values.length ? values : undefined;
+
+      column.setFilter(filtered);
+      onClose();
+    },
+    [column.setFilter, values, onClose]
+  );
+
+  const onClearFilter = useCallback(
+    (event: React.MouseEvent) => {
+      column.setFilter(undefined);
+      onClose();
+    },
+    [column.setFilter, onClose]
+  );
+
+  const clearFilterVisible = useMemo(() => column.filterValue !== undefined, [column.filterValue]);
+
+  const theme = useTheme();
+  const styles = getStyles(theme);
+
+  return (
+    <ClickOutsideWrapper onClick={onCancel} useCapture={true}>
+      <div className={cx(styles.filterContainer)} onClick={stopPropagation}>
+        <VerticalGroup spacing="lg">
+          <VerticalGroup spacing="xs">
+            <Label>Filter by values:</Label>
+            <div className={cx(styles.listDivider)} />
+            <FilterList onChange={setValues} values={values} options={options} />
+          </VerticalGroup>
+          <HorizontalGroup spacing="lg">
+            <HorizontalGroup>
+              <Button size="sm" onClick={onFilter}>
+                Ok
+              </Button>
+              <Button size="sm" variant="secondary" onClick={onCancel}>
+                Cancel
+              </Button>
+            </HorizontalGroup>
+            {clearFilterVisible && (
+              <HorizontalGroup>
+                <Button variant="link" size="sm" onClick={onClearFilter}>
+                  Clear filter
+                </Button>
+              </HorizontalGroup>
+            )}
+          </HorizontalGroup>
+        </VerticalGroup>
+      </div>
+    </ClickOutsideWrapper>
+  );
+};
+
+const getStyles = stylesFactory((theme: GrafanaTheme) => ({
+  filterContainer: css`
+    label: filterContainer;
+    width: 100%;
+    min-width: 250px;
+    height: 100%;
+    max-height: 400px;
+    background-color: ${theme.colors.bg1};
+    border: ${theme.border.width.sm} solid ${theme.colors.border2};
+    padding: ${theme.spacing.md};
+    margin: ${theme.spacing.sm} 0;
+    box-shadow: 0px 0px 20px ${theme.colors.dropdownShadow};
+    border-radius: ${theme.spacing.xs};
+  `,
+  listDivider: css`
+    label: listDivider;
+    width: 100%;
+    border-top: ${theme.border.width.sm} solid ${theme.colors.border2};
+    padding: ${theme.spacing.xs} ${theme.spacing.md};
+  `,
+}));
+
+const calculateUniqueFieldValues = (column: any, noOfColumnFilters: number, field?: Field) => {
+  if (!field) {
+    return {};
+  }
+
+  const set: Record<string, any> = {};
+  const rows = noOfColumnFilters > 1 ? column.filteredRows : column.preFilteredRows;
```

Comment:
I've tried both showing all available distinct rows and all distinct values that are part of filtered rows. I think it's better to show all available rows, but I've left it like this to get some feedback from others.

Reply:
yes, all available rows feels better

Reply:
Yes, I think I'm going to change it back. Thanks for feedback 🙏